src/spikeinterface/core/basesorting.py

```diff
@@ -30,19 +30,26 @@ def __init__(self, sampling_frequency: float, unit_ids: List):
         self._cached_spike_trains = {}
 
     def __repr__(self):
+        return self._repr_header()
+
+    def _repr_header(self, parent_name=None):
```

Comment:
Not sure I like this design. would be better to have an explicit `display_name=True/False`